From d3a79f2666fae7a17eb38c5c8284a048125fe232 Mon Sep 17 00:00:00 2001
From: Bo Shen <voice.shen@atmel.com>
Date: Thu, 26 Sep 2013 10:58:20 +0800
Subject: [PATCH 369/397] ASoC: atmel_ssc_dai: temp solution for sam9g20 audio

Temporary solution for non-dt Linux kernel audio support on sam9g20ek

Signed-off-by: Bo Shen <voice.shen@atmel.com>
---
 sound/soc/atmel/atmel_ssc_dai.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sound/soc/atmel/atmel_ssc_dai.c b/sound/soc/atmel/atmel_ssc_dai.c
index 00881c4..315a3e6 100644
--- a/sound/soc/atmel/atmel_ssc_dai.c
+++ b/sound/soc/atmel/atmel_ssc_dai.c
@@ -455,7 +455,11 @@ static int atmel_ssc_hw_params(struct snd_pcm_substream *substream,
 			| SSC_BF(RCMR_START, start_event)
 			| SSC_BF(RCMR_CKI, SSC_CKI_RISING)
 			| SSC_BF(RCMR_CKO, SSC_CKO_NONE)
+#if defined(CONFIG_MACH_AT91SAM9G20EK) || defined(CONFIG_MACH_AT91SAM9G20EK_2MMC)
+			| SSC_BF(RCMR_CKS, SSC_CKS_CLOCK);
+#else
 			| SSC_BF(RCMR_CKS, SSC_CKS_PIN);
+#endif
 
 		rfmr =	  SSC_BF(RFMR_FSEDGE, SSC_FSEDGE_POSITIVE)
 			| SSC_BF(RFMR_FSOS, SSC_FSOS_NONE)
@@ -470,7 +474,11 @@ static int atmel_ssc_hw_params(struct snd_pcm_substream *substream,
 			| SSC_BF(TCMR_START, start_event)
 			| SSC_BF(TCMR_CKI, SSC_CKI_FALLING)
 			| SSC_BF(TCMR_CKO, SSC_CKO_NONE)
+#if defined(CONFIG_MACH_AT91SAM9G20EK) || defined(CONFIG_MACH_AT91SAM9G20EK_2MMC)
+			| SSC_BF(TCMR_CKS, SSC_CKS_PIN);
+#else
 			| SSC_BF(TCMR_CKS, SSC_CKS_CLOCK);
+#endif
 
 		tfmr =	  SSC_BF(TFMR_FSEDGE, SSC_FSEDGE_POSITIVE)
 			| SSC_BF(TFMR_FSDEN, 0)
-- 
1.8.5.3

